### PR TITLE
Floating button that loads the data in the local DB

### PR DESCRIPTION
**Pull request by Matias-Villalobo**
In this Pr i have done a floating button, that when you press it, it brings the data store in the local db. Also extension functions were implemented in the mapper and in the test of that file.

usecasetest
mvptest
mappertest

https://user-images.githubusercontent.com/73836200/114557853-7afa5c00-9c40-11eb-8f9f-fce1c52c47c5.mp4

### CODE REVIEW

File: app/src/main/java/com/example/memeapp/data/local/LocalMemeDataBaseImpl.kt [hunk 1]
```diff
@@ -1,19 +1,26 @@
 package com.example.memeapp.data.local
```
Comment (MarcosCGlobant) [inline app/src/main/java/com/example/memeapp/data/local/LocalMemeDataBaseImpl.kt]: remove extra line

File: app/src/main/java/com/example/memeapp/data/mapper/MapperExtensions.kt [hunk 2]
```diff
@@ -19,6 +19,21 @@ fun MemesEntity.transformEntityToRealm(): MemeRealmEntity =
...
+    MemesEntity(
+        this.id,
+        this.bottomText,
+        this.image,
+        this.name,
+        this.tag,
+        this.rank,
+        this.topText
+    )
+
```
Comment (MarcosCGlobant) [inline app/src/main/java/com/example/memeapp/data/mapper/MapperExtensions.kt:35]: remove extra line

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -17,4 +17,15 @@
...
+
+    <com.google.android.material.floatingactionbutton.FloatingActionButton
+        android:id="@+id/fab"
+        style="@style/FloatingButtons"
+        android:layout_width="@dimen/fab_circumference_size"
+        android:layout_height="@dimen/fab_circumference_size"
+        android:src="@android:drawable/ic_input_add"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/recyclerView" />
```
Comment (MarcosCGlobant) [inline app/src/main/res/layout/activity_main.xml]:
```suggestion
        app:layout_constraintTop_toBottomOf="@id/recyclerView" />
```
remove the "+" symbol from the constraint

**Comment by BQuinteros** [comment]
Pls for the future try to be more descriptive with the branch. matiasVillalobo/Fab is not a good name for a branch

File: app/src/main/java/com/example/memeapp/presentation/activity/MainActivity.kt [hunk 4]
```diff
@@ -36,4 +39,8 @@ class MainActivity : AppCompatActivity(), ItemClicked {
     override fun memeCardClicked(memeId: Int) {
         presenter.onMemeClicked(memeId)
     }
+
+    private fun loadMemesFromLocalDB() {
```
Comment (SartiMau) [inline app/src/main/java/com/example/memeapp/presentation/activity/MainActivity.kt]: rename this method pls. It can be called setListeners()

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/memeapp/presentation/mvp/presenter/MemeAppPresenter.kt [hunk 5]
```diff
@@ -29,4 +29,13 @@ class MemeAppPresenter(
...
         view.showMemeInfo(memeId)
     }
+
+    override fun getMemesFromDataBase() {
+        view.cleanRecycler()
+        view.showProgressBar(true)
+        model.loadMemesFromDataBase().apply {
+            view.showData(this)
+            view.showProgressBar(false)
+        }
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/memeapp/presentation/mvp/presenter/MemeAppPresenter.kt]: Why you are using a scope function here, if you are not using the model scope?

File: app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -17,4 +17,15 @@
         android:layout_width="match_parent"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
+
+    <com.google.android.material.floatingactionbutton.FloatingActionButton
+        android:id="@+id/fab"
+        style="@style/FloatingButtons"
+        android:layout_width="@dimen/fab_circumference_size"
+        android:layout_height="@dimen/fab_circumference_size"
+        android:src="@android:drawable/ic_input_add"
```
Comment (mauroLonghi) [inline app/src/main/res/layout/activity_main.xml]: put this property into the style

File: app/src/main/java/com/example/memeapp/presentation/mvp/presenter/MemeAppPresenter.kt (hunk 5 above)
Comment (mauroLonghi) [inline app/src/main/java/com/example/memeapp/presentation/mvp/presenter/MemeAppPresenter.kt]: Why you are using a scope function here, if you are not using the model scope?

File: app/src/main/res/values/styles.xml [hunk 7]
```diff
@@ -32,6 +32,13 @@
         <item name="android:layout_margin">@dimen/title_margin_minor_size</item>
     </style>
 
+    <style name="FloatingButtons">
```
Comment (mauroLonghi) [inline app/src/main/res/values/styles.xml]: renamed all the styes, not representt the place witch are using

File: app/src/test/java/com/example/memeapp/data/mapper/MemeMapperTest.kt [hunk 8]
```diff
@@ -36,6 +39,40 @@ class MemeMapperTest {
...
+    fun `transform memes entity to realm Test`() {
+        val meme = MEMES_ENTITY.transformEntityToRealm()
+        assertEquals(MemesConstantsUtils.ID, meme.id)
+        assertEquals(MemesConstantsUtils.NAME, meme.name)
+        assertEquals(MemesConstantsUtils.BOTTOM_TEXT, meme.bottomText)
+        assertEquals(MemesConstantsUtils.IMAGE, meme.image)
+        assertEquals(MemesConstantsUtils.TAG, meme.tag)
+        assertEquals(MemesConstantsUtils.RANK, meme.rank)
+        assertEquals(MemesConstantsUtils.TOP_TEXT, meme.topText)
+    }
```
Comment (mauroLonghi) [inline app/src/test/java/com/example/memeapp/data/mapper/MemeMapperTest.kt]: you can use a method to reuse the functionality

File: app/src/main/java/com/example/memeapp/data/local/LocalMemeDataBaseImpl.kt (hunk 1 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/memeapp/data/local/LocalMemeDataBaseImpl.kt]: Done thx

File: app/src/main/java/com/example/memeapp/data/mapper/MapperExtensions.kt (hunk 2 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/memeapp/data/mapper/MapperExtensions.kt:35]: Done thx

File: app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (Matias-Villalobo) [reply app/src/main/res/layout/activity_main.xml]: Done thx

File: app/src/main/java/com/example/memeapp/presentation/activity/MainActivity.kt (hunk 4 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/memeapp/presentation/activity/MainActivity.kt]: Done thx

File: app/src/main/res/values/styles.xml (hunk 7 above)
Comment (Matias-Villalobo) [reply app/src/main/res/values/styles.xml]: Done thx

File: app/src/test/java/com/example/memeapp/data/mapper/MemeMapperTest.kt (hunk 8 above)
Comment (Matias-Villalobo) [reply app/src/test/java/com/example/memeapp/data/mapper/MemeMapperTest.kt]: Done thx

**Comment by Matias-Villalobo** [comment]
As you can see in the image below, i have changed the icon of the floating button

image

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/example/memeapp/presentation/mvp/presenter/MemeAppPresenter.kt (hunk 5 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/memeapp/presentation/mvp/presenter/MemeAppPresenter.kt]: now its no more, thx

**Review by MarcosCGlobant — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

File: app/src/main/res/layout/activity_main.xml (hunk 6 above)
Comment (Matias-Villalobo) [reply app/src/main/res/layout/activity_main.xml]: Done thx

File: app/src/main/java/com/example/memeapp/presentation/mvp/presenter/MemeAppPresenter.kt (hunk 5 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/memeapp/presentation/mvp/presenter/MemeAppPresenter.kt]: Now its no more, thx

**Review by lucianoeli — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Pull request merged**